File: src/main/java/com/mrkirby153/snowsgivingbot/commands/slashcommands/AdminSlashCommands.java
package com.mrkirby153.snowsgivingbot.commands.slashcommands;

import com.mrkirby153.botcore.command.slashcommand.SlashCommand;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import me.mrkirby153.kcutils.Time;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import org.springframework.stereotype.Service;

@Slf4j
@Service
@RequiredArgsConstructor
public class AdminSlashCommands {

    @SlashCommand(name = "ping", description = "Checks the bot's ping")
    public void ping(SlashCommandEvent event) {
        long start = System.currentTimeMillis();
        event.deferReply(true).queue(hook -> {
            hook.editOriginal("Pong! " + Time.format(1, System.currentTimeMillis() - start))
                .queue();
        });
    }
}

File: src/main/kotlin/com/mrkirby153/giveaways/commands/AdminSlashCommands.kt
package com.mrkirby153.giveaways.commands

import com.mrkirby153.botcore.command.slashcommand.dsl.DslCommandExecutor
import com.mrkirby153.botcore.command.slashcommand.dsl.slashCommand
import com.mrkirby153.botcore.coroutine.await
import me.mrkirby153.kcutils.Time
import org.springframework.stereotype.Component

@Component
class AdminSlashCommands : DslSlashCommandProvider {
    override fun register(executor: DslCommandExecutor) {
        executor.registerCommands {
            slashCommand("ping") {
                description = "Checks the bot's ping"
                run {
                    val start = System.currentTimeMillis()
                    val hook = deferReply(true).await()
                    hook.editOriginal(
                        "Pong! Took ${
                            Time.format(
                                1,
                                System.currentTimeMillis() - start
                            )
                        }"
                    ).await()
                }
            }
        }
    }
}